diff --git a/app/pocket_acq/makefile b/app/pocket_acq/makefile
index ca21c75..d1d6bcb 100644
--- a/app/pocket_acq/makefile
+++ b/app/pocket_acq/makefile
@@ -19,9 +19,9 @@ else ifeq ($(shell uname -sm),Darwin arm64)
              -lfftw3f -lpthread -lm
     OPTIONS = -I/opt/homebrew/include
 else
-    CC = g++
+    CC ?= g++
     INCLUDE = -I$(SRC) -I$(LIB)/RTKLIB/src
-    LDLIBS = $(LIB)/linux/libsdr.a $(LIB)/linux/librtk.a -lfftw3f -lpthread -lm
+    LDLIBS = -lfec -lldpc -lsdr -lfftw3f -lrtk -lpthread -lm -lusb-1.0
     OPTIONS =
 endif
 
diff --git a/app/pocket_conf/makefile b/app/pocket_conf/makefile
index 1251764..4bb6332 100644
--- a/app/pocket_conf/makefile
+++ b/app/pocket_conf/makefile
@@ -2,7 +2,7 @@
 #  makefile for pocket_conf
 #
 
-CC = g++
+CC ?= g++
 
 SRC = ../../src
 LIB = ../../lib
@@ -19,7 +19,7 @@ else ifeq ($(shell uname -sm),Darwin arm64)
 else
     INCLUDE = -I$(SRC) -I$(LIB)/RTKLIB/src
     OPTIONS =
-    LDLIBS = $(LIB)/linux/libsdr.a -lusb-1.0 -lpthread
+    LDLIBS = -lfec -lldpc -lsdr -lfftw3f -lrtk -lusb-1.0 -lpthread
 endif
 
 WARNOPT = -Wall -Wextra -Wno-missing-field-initializers -Wno-unused-parameter
diff --git a/app/pocket_dump/makefile b/app/pocket_dump/makefile
index 82c6c8e..bb9022a 100644
--- a/app/pocket_dump/makefile
+++ b/app/pocket_dump/makefile
@@ -17,10 +17,10 @@ else ifeq ($(shell uname -sm),Darwin arm64)
     OPTIONS = -Wno-deprecated
     LDLIBS = -L/opt/homebrew/lib $(LIB)/macos/libsdr.a -lusb-1.0 -lpthread
 else
-    CC = g++
+    CC ?= g++
     INCLUDE = -I$(SRC) -I$(LIB)/RTKLIB/src
     OPTIONS =
-    LDLIBS = $(LIB)/linux/libsdr.a -lusb-1.0 -lpthread
+    LDLIBS = -lfec -lldpc -lsdr -lfftw3f -lrtk -lusb-1.0 -lpthread
 endif
 
 WARNOPT = -Wall -Wextra -Wno-missing-field-initializers -Wno-unused-parameter
diff --git a/app/pocket_scan/makefile b/app/pocket_scan/makefile
index 5c3ca70..b1d3e72 100644
--- a/app/pocket_scan/makefile
+++ b/app/pocket_scan/makefile
@@ -2,7 +2,7 @@
 #  makefile for pocket_scan
 #
 
-CC = g++
+CC ?= g++
 
 SRC = ../../src
 LIB = ../../lib
diff --git a/app/pocket_snap/makefile b/app/pocket_snap/makefile
index b43ad5b..b49ac96 100644
--- a/app/pocket_snap/makefile
+++ b/app/pocket_snap/makefile
@@ -18,10 +18,10 @@ else ifeq ($(shell uname -sm),Darwin arm64)
              -lfftw3f -lpthread -lm
     OPTIONS = -I/opt/homebrew/include
 else
-    CC = g++
+    CC ?= g++
     INCLUDE = -I$(SRC) -I$(LIB)/RTKLIB/src
-    LDLIBS = $(LIB)/linux/libsdr.a $(LIB)/linux/librtk.a -lfftw3f -lpthread -lm
-    OPTIONS = -DAVX2 -mavx2 -mfma
+    LDLIBS = -lfec -lldpc -lsdr -lfftw3f -lrtk -lpthread -lm -lusb-1.0
+    OPTIONS = # -DAVX2 -mavx2 -mfma
 endif
 ifeq ($(shell uname -m),aarch64)
     OPTIONS = -DNEON
diff --git a/app/pocket_trk/makefile b/app/pocket_trk/makefile
index 0b33529..2db616c 100644
--- a/app/pocket_trk/makefile
+++ b/app/pocket_trk/makefile
@@ -2,7 +2,7 @@
 #  makefile for pocket_trk, fftw_wisdom
 #
 
-CC = g++
+CC ?= g++
 
 SRC = ../../src
 LIB = ../../lib
@@ -23,9 +23,8 @@ else ifeq ($(shell uname -sm),Darwin arm64)
     OPTIONS = -Wno-deprecated
 else
     INCLUDE = -I$(SRC) -I$(LIB)/RTKLIB/src
-    LIBSDR = $(LIB)/linux/libsdr.a
-    LDLIBS = $(LIBSDR) $(LIB)/linux/librtk.a $(LIB)/linux/libfec.a \
-             $(LIB)/linux/libldpc.a -lfftw3f -lpthread -lm -lusb-1.0
+    LIBSDR = 
+    LDLIBS = -lfec -lldpc -lsdr -lfftw3f -lrtk -lpthread -lm -lusb-1.0
     OPTIONS =
 endif
 
@@ -38,7 +37,7 @@ TARGET = pocket_trk fftw_wisdom
 
 all: $(TARGET)
 
-pocket_trk: pocket_trk.o $(LIBSDR)
+pocket_trk: pocket_trk.o
 fftw_wisdom: fftw_wisdom.o
 
 pocket_trk.o: $(SRC)/pocket_sdr.h
diff --git a/lib/build/librtk.mk b/lib/build/librtk.mk
index ac4fec6..fc1dce7 100644
--- a/lib/build/librtk.mk
+++ b/lib/build/librtk.mk
@@ -15,7 +15,7 @@ else ifeq ($(shell uname -sm),Darwin arm64)
     OPTIONS= -DSVR_REUSEADDR -DTRACE -DMACOS -Wno-deprecated
     LDLIBS =
 else
-    CC = gcc
+    CC ?= gcc
     INSTALL = ../linux
     OPTIONS= -DSVR_REUSEADDR -DTRACE
     LDLIBS =
diff --git a/lib/build/libsdr.mk b/lib/build/libsdr.mk
index cafabbe..53beecd 100644
--- a/lib/build/libsdr.mk
+++ b/lib/build/libsdr.mk
@@ -24,12 +24,11 @@ else ifeq ($(shell uname -sm),Darwin arm64)
     LDLIBS = -L/opt/homebrew/lib ./librtk.a ./libfec.a ./libldpc.a -lfftw3f \
              -lusb-1.0 -lpthread
 else
-    CC = g++
+    CC ?= g++
     INSTALL = ../linux
     INCLUDE = -I$(SRC) -I../RTKLIB/src
-    OPTIONS = -DAVX2 -mavx2 -mfma
-    LDLIBS = ./librtk.a ./libfec.a ./libldpc.a -lfftw3f -lpthread -lusb-1.0 -lm \
-             -lpthread
+    OPTIONS = # -DAVX2 -mavx2 -mfma
+    LDLIBS = -lfftw3f -lpthread -lusb-1.0 -lm -lpthread
 endif
 ifeq ($(shell uname -m),aarch64)
     OPTIONS = -DNEON
@@ -47,7 +46,7 @@ TARGET = libsdr.so libsdr.a
 all : $(TARGET)
 
 libsdr.so: $(OBJ)
-	$(CC) -shared -o $@ $(OBJ) $(LDLIBS)
+	$(CC) -shared -o $@ $(OBJ)
 
 libsdr.a: $(OBJ)
 	$(AR) r $@ $(OBJ)
