Index: uboot-2020.10.bak/arch/arm/dts/rk3399.dtsi
===================================================================
--- uboot-2020.10.bak.orig/arch/arm/dts/rk3399.dtsi
+++ uboot-2020.10.bak/arch/arm/dts/rk3399.dtsi
@@ -248,7 +248,7 @@
 				<0 0 0 3 &pcie0_intc 2>,
 				<0 0 0 4 &pcie0_intc 3>;
 		linux,pci-domain = <0>;
-		max-link-speed = <1>;
+		max-link-speed = <2>;
 		msi-map = <0x0 &its 0x0 0x1000>;
 		phys = <&pcie_phy 0>, <&pcie_phy 1>,
 		       <&pcie_phy 2>, <&pcie_phy 3>;
