From 67f2cd95cf2e92eee21a4be1dc2cf52e8fd94fa2 Mon Sep 17 00:00:00 2001
From: marionm <mathis.marion@grenoble-inp.org>
Date: Fri, 20 Aug 2021 09:52:22 +0200
Subject: [PATCH] Uninitialized variable

Signed-off-by: marionm <mathis.marion@grenoble-inp.org>
---
 software/lib/libmockturtle.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/software/lib/libmockturtle.c b/software/lib/libmockturtle.c
index 7019146..9a0ddcd 100644
--- a/software/lib/libmockturtle.c
+++ b/software/lib/libmockturtle.c
@@ -521,7 +521,7 @@ int trtl_cpu_load_application_raw(struct trtl_dev *trtl,
 	struct trtl_desc *wdesc = (struct trtl_desc *)trtl;
 	char path[TRTL_PATH_LEN + TRTL_NAME_LEN + 1 + 3];
 	int fd;
-	ssize_t ret;
+	ssize_t ret = 0;
 	size_t i = 0;
 
 	snprintf(path, sizeof(path), "%s/%s-%02u",
-- 
2.30.2

